cursor/project-overview.md

```diff
@@ -0,0 +1,39 @@
+
+There are 2 services, an inventory service (pkg/lib/inventory) that gets the data from the vm and a validation service (validation/service) that makes sure the the data is compatible for migration (example an unsupported file system may not be able to be migrated).
+
+After the admin configures the provider, network and storage CR-they can create a plan CR for the migration plan. They will choose which groups of providers can be migrated together and choose which ones will have which storage and mapping.
+
+There are different types of migrations (cold, warm, live, conversion, storage). All migration types support RCM or RawCopyMode. RawCopyMode shows up in the plan CR as "skipGuestConversion: true" meaning it will not use virtv2v to install virtio drivers. VDDK is required for RCM. VDDK is VMware's library that gives access to the the VM disk files or VDMK. VDDK is also needed for warm migrations and CDI storage migrations. Migration progress is monitored in (pkg/monitoring)
+
+  -cold migration is the default migration where the VM is turned off before the migration. This has a longer downtime. In a cold migration a DataVolume is created and virtv2v does the guest conversion by swapping proprietary drivers with its own drivers and then it goes to the target vm.
+
+ -warm migration has shorter downtime because the VM stays on during the migration. There is one snapshot taken that is copied, then a series of snapshots where only the changes between snapshots are copied. Then finally the VM is turned off for the cutover phase and the final changes are copied-target vm is made and the guest conversion moves this copy to the target vm. This is supported for vSphere, oVirt and Red Hat virtualization.
+
+ -Live migration has almost no downtime. This is for moving data from one cluster to another. An empty DataVolume is created in addition to a standby VM on the target cluster. Then KubeVirt migrates the storage and memory and VirtualMachineInstanceMigration manages the state.
+
+ -Conversion only migration does NOT migrate data. It installs the virtIO drivers to change the guest OS and puts the changed guest OS in a target vm.
+
+ -storage migration copies VM disk data to a persistent volume claim PVC in openshift that can be used by a KubeVirt VM. In the final step of the migration, the controller creates a virtual machine CR and a vm pod/virt-launcher and the transferred PVCs are in the format of disks on the VM (pkg/controller/plan/adapter). These are the formats needed for for a storage migration for each provider: 
```

Comment:
What is a storage migration? I only see cold/warm/live/onlyconversion in the forklift MigrationType list. They should all result in a PVC attached to a VM.